expect_test.go

```diff
@@ -77,11 +77,18 @@ func newTestConsole(t *testing.T, opts ...ConsoleOpt) (*Console, error) {
 
 func expectNoError(t *testing.T) ConsoleOpt {
 	return WithExpectObserver(
-		func(matcher Matcher, buf string, err error) {
-			if matcher == nil && err != nil {
+		func(matchers []Matcher, buf string, err error) {
+			if err == nil {
+				return
+			}
+			if len(matchers) == 0 {
 				t.Fatalf("Error occured while matching %q: %s\n%s", buf, err, string(debug.Stack()))
-			} else if err != nil {
-				t.Fatalf("Failed to find %v in %q: %s\n%s", matcher.Criteria(), buf, err, string(debug.Stack()))
+			} else {
+				matching := fmt.Sprintf("%q", matchers[0].Criteria())
+				for _, matcher := range matchers[1:] {
+					matching += fmt.Sprintf(", %q", matcher.Criteria())
+				}
```

Comment:
nit, prefer building `var criterias []string` and `strings.Join(criterias, ", ")` over string concatentation.

Reply:
Okay, I populate criterias now and run strings.Join.